data/moves.js

```diff
@@ -5462,11 +5462,13 @@ exports.BattleMovedex = {
 		priority: 0,
 		flags: {protect: 1, reflectable: 1, heal: 1, mystery: 1},
 		onHit: function (target) {
+			let healResult = false;
 			if (this.isTerrain('grassyterrain')) {
-				this.heal(this.modify(target.maxhp, 0.667)); // TODO: find out the real value
+				healResult = this.heal(this.modify(target.maxhp, 0.667)); // TODO: find out the real value
 			} else {
-				this.heal(Math.ceil(target.maxhp * 0.5));
+				healResult = this.heal(Math.ceil(target.maxhp * 0.5));
 			}
+			return healResult;
```

Comment:
It does not need to be this hard.

Just replace `this.heal` with `return this.heal` and nothing else needs to be done.

Reply:
Oh, yeah, that does make sense.

Reply:
Huh, it was that easy? Why does `scripts.js` go to such great lengths when it processes healing?

Reply:
https://github.com/Zarel/Pokemon-Showdown/blob/1e169428cace3a87fe186b954b892fa93c389f6f/data/scripts.js#L666-L675

Doesn't look like much more "great lengths". It manually handles failing if the heal fails, but that's about it.

Reply:
Well, that, and the weird change in mechanics before Gen 5, and the fact that it doesn't seem to trigger any Heal events...